docs/git.md

```diff
@@ -0,0 +1,77 @@
+# Git Guide
+
+## What?
+Git is a version control system. 
+It allows teams to collaborate on a project. Git is considered to be the industry standard 
+version control system.
+
+## Guide
+
+### 1) Setup
+* macos: `brew install git`
+* ubuntu: `sudo apt-get install git`
+* fedora: `sudo dnf install git`
+* windows: download from https://gitforwindows.org/ and run the installer
+
+### 2) Basic configuration
+* set global name and email (optional)
+
+
+    git config --global user.name "John Doe"
+
+
+    git config --global user.email "john.doe@example.com"
+
+
+All commits will show the name and email configured. This will apply for all repositories on your computer.
+
+* set local name and email (recommended)
+    
+
+    git config user.name "John Doe"
+
+
+    git config user.email "john.doe@example.com"
+
+Same as the global one but it only applies to the current directory.
+This will make it possible to configure different emails/names for different projects(=clients).
+
+### 3) basic commands
+Your IDE does a lot for you but it pays to know the cli commands.
+
+* `git fetch` fetches the latest state of the branches. You then know if you need to pull any changes
+* `git status` shows the files which are staged for commit, shows files which aren't added to git and shows if the current branch is behind/ahead of the remote branch on the server.
+* `git branch` makes a new branch from the current branch
+* `git checkout` checks out a specific branch
+* `git pull` pulls the latest commits on the server into your branch.
+* `git add {path}` stages files to be committed.
+* `git commit -m "{message}"` commits the files which are staged for commit.
+* `git push` pushes the local commits to the remote branch.
+
+### 4) Advanced commands
+* `git push --force-with-lease` this force pushes the current branch with rewritten history to the remote. (dangerous if you don't know what you are doing)
+* `git rebase` (will be explained in a section below)
```

Comment:
Refers to "section below" but not really found any content related to git rebase (apart from the external resources).

Reply:
added that section, it was a bit late when I wrote this page :)